pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -477,50 +447,105 @@ func TestReconcile(t *testing.T) {
+				ExpectedMachines:    IntPtr(1),
+				CurrentHealthy:      IntPtr(1),
+				RemediationsAllowed: 1,
+				Conditions: mapiv1beta1.Conditions{
+					remediationAllowedCondition,
 				},
-			}
-			result, err := r.Reconcile(ctx, request)
-			assertEvents(t, tc.testCase, tc.expectedEvents, recorder.Events)
-			if tc.expected.error != (err != nil) {
-				var errorExpectation string
-				if !tc.expected.error {
-					errorExpectation = "no"
-				}
-				t.Errorf("Test case: %s. Expected: %s error, got: %v", tc.node.Name, errorExpectation, err)
-			}
+			},
+		},
 
-			if result != tc.expected.result {
-				if tc.expected.result.Requeue {
-					before := tc.expected.result.RequeueAfter
-					after := tc.expected.result.RequeueAfter + time.Second
-					if after < result.RequeueAfter || before > result.RequeueAfter {
-						t.Errorf("Test case: %s. Expected RequeueAfter between: %v and %v, got: %v", tc.node.Name, before, after, result)
-					}
-				} else {
-					t.Errorf("Test case: %s. Expected: %v, got: %v", tc.node.Name, tc.expected.result, result)
-				}
-			}
+		{ //When remediationTemplate is set and node transitions to unhealthy, new Remediation Request should be created
+			name:                        "create new external remediation",
+			machine:                     machineWithNodeUnHealthy,
+			node:                        nodeUnHealthy,
+			mhc:                         mhcWithRemediationTemplate,
+			externalRemediationMachine:  nil,
+			externalRemediationTemplate: ermTemplate,
+			expected: expectedReconcile{
+				result: reconcile.Result{},
+				error:  false,
+			},
+			expectedEvents: []string{},
+			expectedStatus: &mapiv1beta1.MachineHealthCheckStatus{
+				ExpectedMachines:    IntPtr(1),
+				CurrentHealthy:      IntPtr(0),
+				RemediationsAllowed: 0,
+				Conditions: mapiv1beta1.Conditions{
+					remediationAllowedCondition,
+				},
+			},
+		},
+
+		{ //When remediationTemplate is set and node transitions to unhealthy, and a a Remediation Request already exist
+			name:                        "external remediation is in process",
+			machine:                     machineWithNodeUnHealthy,
+			node:                        nodeUnHealthy,
+			mhc:                         mhcWithRemediationTemplate,
+			externalRemediationMachine:  erm,
+			externalRemediationTemplate: ermTemplate,
+			expected: expectedReconcile{
+				result: reconcile.Result{},
+				error:  false,
+			},
+			expectedEvents: []string{},
+			expectedStatus: &mapiv1beta1.MachineHealthCheckStatus{
+				ExpectedMachines:    IntPtr(1),
+				CurrentHealthy:      IntPtr(0),
+				RemediationsAllowed: 0,
+				Conditions: mapiv1beta1.Conditions{
+					remediationAllowedCondition,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			recorder := record.NewFakeRecorder(2)
+			r := newFakeReconcilerWithCustomRecorder(recorder, buildRunTimeObjects(tc)...)
+			assertBaseReconcile(t, tc, ctx, r)
+			//assertExternalRemediation
```

Comment:
What's this comment? Do we need this?

Reply:
Nope. it's quite redundant.
I've removed it.